rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/rest/TbHttpClient.java

```diff
@@ -242,12 +242,16 @@ public URI buildEncodedUri(String endpointUrl) {
         return uri;
     }
 
-    private String getData(TbMsg msg) {
-        String data = msg.getData();
+    private String getData(TbMsg tbMsg) {
+        return parseJsonStringToPlainText(tbMsg.getData(), config.isParseToPlainText());
+    }
 
-        if (config.isTrimDoubleQuotes()) {
+    protected String parseJsonStringToPlainText(String data, boolean parseToJson) {
```

Comment:
it looks like the method doing something only for parseToJson == true
It makes sense to move such a case upper and not fire the method if parseToJson == false